Server/AutomationActions/Query/Specifiers/QuerySpecifierByPredicate.m

```diff
@@ -0,0 +1,83 @@
+    
+    if (range.location != NSNotFound) {
+        NSLog(@"Predicate '%@' contains elementType. Proceeed with replacement of XCUIElementType String value on int value.", actualPredicateString);
+
+        NSString *matchPattern = @"elementType == '(\\w+)'";
+        NSRange matchRange = [actualPredicateString rangeOfString:matchPattern options:NSRegularExpressionSearch];
+
+        if (matchRange.location != NSNotFound) {
+            NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:matchPattern options:0 error:nil];
+            NSString *actualMatch = [actualPredicateString substringWithRange:matchRange];
+            NSString *replacementString = [self replacementStringForElementType:actualMatch];
+            
+            resultPredicateString = [regex stringByReplacingMatchesInString:actualPredicateString options:0 range:NSMakeRange(0, [actualPredicateString length]) withTemplate:replacementString];
+            NSLog(@"Replace original predicate '%@' with the replacement: %@", actualPredicateString, resultPredicateString);
+        } else {
+            @throw [CBXException withFormat:@"Can not parse element type. Actual string: '%@'. Expected type predicate form as: elementType == 'Button'", self.value];
+        }
+    } else {
+        resultPredicateString = actualPredicateString;
+        NSLog(@"Predicate does not contain elementType attribute. Proceeed with the original value: %@", resultPredicateString);
+    }
+    
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:resultPredicateString];
+    
+    return [query matchingPredicate:predicate];
+}
+
+/*
+ * Replace any occurences of XCUIElementType from string value with single quotes into int value.
+ * The method return the original string with this replacement
+ *
+ * For example:
+ * - incoming method argument as simple predicate of String type: original = "elementType == 'StaticText'"
+ * - returns replacement string as predicate as well: replacementString = "elementType == 48"
+ */
+- (NSString *)replacementStringForElementType:(NSString *)original {
+    // Find any text within single quotes
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@"'(.*)'" options:0 error:NULL];
+    NSTextCheckingResult *match = [regex firstMatchInString:original options:0 range:NSMakeRange(0, [original length])];
+    NSString *shortTypeName = [original substringWithRange:[match rangeAtIndex:1]];
+    NSLog(@"Found element type '%@' in predicate string: %@", shortTypeName, original);
+    
+    // Replace String type occurence on int
+    NSUInteger type = [JSONUtils elementTypeForString:shortTypeName];
+    NSLog(@"Replace element type '%@' into int: %i", shortTypeName, (unsigned int)type);
+    
+    // Provide new simple predicate with int element type
+    NSString *replacementString = [NSString stringWithFormat:@"elementType == %i", (unsigned int)type];
```

Comment:
Always use auto-boxing to print numbers.

```suggestion
    NSString *replacementString = [NSString stringWithFormat:@"elementType == %@", @(type)];
```

This avoids casting and it future proofs against changes in type (recall the transition from 32 bit int to 64 bit int).